scipy/stats/_hypotests.py

```diff
@@ -892,27 +892,50 @@ def barnard_exact(table, alternative="two-sided", pooled=True, n_iter=1):
         )
         raise ValueError(msg)
 
-    p_value = _binomial_maximisation_of_p_value_with_nuisance_param(
-        total_col_1, total_col_2, index_arr, n_iter
+    get_pvalue_from = _binomial_maximisation_of_p_value_with_nuisance_param(
+        total_col_1, total_col_2, index_arr
     )
+
+    result = shgo(
+        get_pvalue_from,
+        bounds=((0, 1),),
+        n=32,  # Need to be a power of two since it is used by sobol
+        sampling_method="sobol",
+        iters=n_iter,
```

Comment:
Also for `n` I am still wondering if it should be accessible. I guess it's fine since we can always increase the number of iterations if needed. From my tests, I could not say that I really needed to touch this.

```suggestion
        n=32,  # Need to be a power of two since it is used by sobol
        sampling_method="sobol",
        minimizer_kwargs={"method": "Nelder-Mead"},
        iters=n_iter,
```